UKHomeOffice/sgmr-service
# SPCR-172 Cypress & related updates

## Description

To resolve a few vulnerabilities we needed to update Cypress and other test related packages
This PR includes
- Update Cypress, major version steps, to 9.13
- Update mocha
- Update mochawesome
- Update mochawesome-report-generator
- update cypress-axe

## To test

- ask Jen for the `apiKey`
- remove your package-lock.json and node-modules
- run `npm i`
- run `npm run cypress:test:report -- -b chrome -s cypress/integration/sGMR/user-register.spec.js`
- > tests for this file should run in the terminal and generate a report, it will tell you the location
- find and view the report
- > report should be viewable
- run `npm run cypress:runner -- --env configFile=dev`
- > Cypress runner should open
- run `user-register` tests
- > tests should run

Note: We expect most Cypress tests to fail at the moment as they need updating, but they should run then fail as opposed to not run


## Developer Checklist

\* Required






Ensure you delete the branch once the Pull Request is merged.
